apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -301,6 +453,209 @@ impl UsageReportingGenerator<'_> {
             }
         }
     }
+
+    fn generate_extended_references(&mut self) -> ExtendedReferenceStats {
+        self.fragment_spread_set.clear();
+        self.enums_by_name.clear();
+        self.input_field_references.clear();
+
+        if let Ok(operation) = self
+            .references_doc
+            .get_operation(self.operation_name.as_deref())
+        {
+            self.process_extended_refs_for_selection_set(&operation.selection_set);
+        }
+
+        ExtendedReferenceStats {
+            referenced_input_fields: self.input_field_references.clone(),
+            referenced_enums: self.enums_by_name.clone(),
+        }
+    }
+
+    fn add_enum_reference(&mut self, enum_name: String, enum_value: String) {
+        self.enums_by_name
+            .entry(enum_name)
+            .or_default()
+            .insert(enum_value.to_string());
+    }
+
+    fn add_input_object_reference(
+        &mut self,
+        type_name: String,
+        field_name: String,
+        is_referenced: bool,
+        is_null_reference: bool,
+    ) {
+        match self
+            .input_field_references
+            .entry(type_name)
+            .or_default()
+            .entry(field_name)
```

Comment:
if we're in a hot path here. We already spotted that using the entry api might be less performant

Reply:
I'm not sure if you meant that I shouldn't use the entry API at all or if I should only avoid using it unnecessarily in an inner loop, so I've changed it to never use the entry API. Let me know if I should change it so we still use the entry API but only once in the outer loop.